##### Вариант 28
###### В приведенной таблице даны значения урожайности картофеля (тыс. тонн с гектара) в зависимости от сорта картофеля (фактор А) и типа примененного удобрения (фактор В). 
##### С помощью двухфакторного дисперсионного анализа без повторных измерения выяснить:
###### а. значимы ли различия в средней урожайности различных сортов картофеля независимо от типа удобрения;
###### б. значимо ли влияние типа применяемого удобрения на урожайность независимо от сорта.

###### |----|--**B1**--|--**B2**--|--**B3**--|--**B4**--|
###### | **A1**| 5.265  | 6.998  | 6.032  | 8.851  |
###### | **A2**| 9.656  | 7.239  | 7.552  | 9.684  |
###### | **A3**| 9.16   | 8.084  | 10.504 | 9.045  |

In [1]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
import numpy as np

In [18]:
data = {
    'B1': [5.265, 9.656, 9.16],
    'B2': [6.998, 7.239, 8.084],
    'B3': [6.032, 7.552, 10.504],
    'B4': [8.851, 9.684, 9.045]
}

df = pd.DataFrame(data, index=['A1', 'A2', 'A3']).reset_index()
df = df.melt(id_vars='index', var_name='B', value_name='Yield')
df.rename(columns={'index': 'A'}, inplace=True)

df.head()

A   B  Yield
0  A1  B1  5.265
1  A2  B1  9.656
2  A3  B1  9.160
3  A1  B2  6.998
4  A2  B2  7.239

###### Проведение двухфакторного дисперсионного анализа c помощью модели ANOVA без взаимодействия

In [22]:
model = ols('Yield ~ C(A) + C(B)', data=df).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)

             sum_sq   df         F    PR(>F)
C(A)      12.411756  2.0  3.537514  0.096633
C(B)       4.859507  3.0  0.923349  0.484536
Residual  10.525830  6.0       NaN       NaN


##### Выводы

##### 1. Влияние фактора A (Сорт картофеля):
###### Сумма квадратов (sum_sq): 12.411756
###### Степени свободы (df): 2.0
###### F-статистика (F): 3.537514
###### p-значение (PR(>F)): 0.096633
###### p-значение для фактора A (0.096633) больше стандартного уровня значимости 0.05, что указывает на то, что различия в средней урожайности между сортами картофеля не являются статистически значимыми на уровне 5%. Однако, p-значение близко к этому уровню.

##### 2. Влияние фактора B (Тип удобрения):
###### Сумма квадратов (sum_sq): 4.859507
###### Степени свободы (df): 3.0
###### F-статистика (F): 0.923349
###### p-значение (PR(>F)): 0.484536
###### p-значение для фактора B (0.484536) значительно больше 0.05, что указывает на то, что различия в средней урожайности между типами удобрений не являются статистически значимыми.

##### 3. Остаточная дисперсия:
###### Сумма квадратов (Residual): 10.525830
###### Степени свободы (df): 6.0
###### Остаточная дисперсия показывает, что есть вариация в данных, которая не объясняется факторами A и B.

##### Итог
###### В выборке не было найдено статистически значимых различий в урожайности картофеля в зависимости от сорта и типа удобрения. Фактор A показывает тенденцию к значимости. Фактор B не показал значимого влияния на урожайность.